project/src/main/java/com/google/sps/data/Places.java

```diff
@@ -45,6 +47,23 @@
     return ImmutableList.copyOf(mutablePlaces);
   }
 
+  /**
+   * Sorts the given list of Places by score.
+   * @param places the list we want to sort.
+   * @param userLocation the user's loaction, used for scores calculations
+   * @return a new list containing the original list's elements
+   *     arranged by descending order of their scores.
+   */
+  public static ImmutableList<Place> scoreSort(ImmutableList<Place> places, LatLng userLocation) {
+    List<Place> mutablePlaces = new ArrayList<>(places);
+    Map<Place, Double> placesScores = new PlacesScorer(places, userLocation).getScores();
+    Collections.sort(mutablePlaces, (p1, p2) -> {
+      return placesScores.get(p2).compareTo(placesScores.get(p1));
+    });
+    return ImmutableList.copyOf(mutablePlaces);
+  }
+
+
   /**
```

Comment:
Do we maybe want a secondary sort in case of a tie? by name or something else?

Reply:
I think that since Places API returns us their results in the order that they think we will prefer (as documented on Google Places API) , it is best in case of a tie to preserve the original order.

Reply:
SGTM